runtime/sema/variable_activations.go

```diff
@@ -271,19 +271,19 @@ func (a *VariableActivations) Depth() int {
 	return len(a.activations)
 }
 
-type variableDeclaration struct {
-	identifier               string
-	ty                       Type
-	docString                string
-	access                   ast.Access
-	kind                     common.DeclarationKind
-	pos                      ast.Position
-	isConstant               bool
-	argumentLabels           []string
-	allowOuterScopeShadowing bool
+type VariableDeclaration struct {
```

Comment:
Instead of exporting this low-level type (it is used for any sema variable, values and types), keep it unexported. Then, instead of having a checker `contractVariableHandler`, have a `contractValueHandler`, i.e. have `ContractValueHandlerFunc` return a `sema.ValueDeclaration`. This basically reuses the existing mechanism of declaring values in a checker, which is also used by the runtime environment to declare standard library values.

Reply:
The problem with that is `stdlib.StandardLibraryValue` (implementation of `sema.ValueDeclaration`) also needs the interpreter value. But at the checker level, we don't have enough info to construct the value (i.e: constructor/host-function generator, etc.). The value is provided by the `InterpreterConfig.ContractValueHandler`

Reply:
Yeah, `stdlib.StandardLibraryValue` is just a convenience that can be used for both sema and interpreter value declarations. 

In you case you can just leave the `Value` field empty, it's not used by the checker.

Reply:
@SupunS OK to push a commit? Can also open a PR against this one if you'd like?

Reply:
Sure, please feel free to directly push

Reply:
Reverted the export in [cbf628a](https://github.com/onflow/cadence/pull/1923/commits/cbf628af6e4bb039ce7dba82949fdc13dff0f3c0) and also cleaned up a bit along the way by un-exporting some other functions that only sema internally should call